src/sync/deployment.ts

```diff
@@ -0,0 +1,73 @@
+import { GitHubAPI } from "probot";
+import { Repository } from "models/subscription";
+import { WebhookPayloadDeploymentStatus } from "@octokit/webhooks";
+import { GitHubInstallationClient } from "../github/client/github-installation-client";
+import { LoggerWithTarget } from "probot/lib/wrap-logger";
+import { transformDeployment } from "../transforms/transform-deployment";
+
+const fetchDeployments = async (gitHubInstallationClient: GitHubInstallationClient, repository: Repository, cursor?: string | number, perPage?: number) => {
+	const deploymentData = await gitHubInstallationClient.getDeploymentsPage(repository.owner.login, repository.name, perPage, cursor);
+	const edges = deploymentData.repository.deployments.edges;
+	const deployments = edges?.map(({ node: item }) => item) || [];
+
+	return {
+		edges,
+		deployments
+	};
+};
+
+const getTransformedDeployments = async (deployments, _github: GitHubAPI, gitHubInstallationClient: GitHubInstallationClient, jiraHost: string, logger: LoggerWithTarget) => {
+	const transformTasks = await deployments.reduce(async (acc, current) => {
```

Comment:
I don't recommend mixing awaits with reduce.  Reduce is complex enough as is without having to think about promises

Reply:
instead, you probably want to do a `await Promise.all(deployments.map(...))`

Reply:
yes, agree was replacing a map.filter, but the reduce/promises do make it unnecessarily more complicated.

Reply:
Simplified!